core/src/consensus/tendermint/worker.rs

```diff
@@ -1107,31 +1111,25 @@ impl Worker {
 
         let header = sealed_block.header();
 
-        if let TendermintState::ProposeWaitBlockGeneration {
-            parent_hash: expected_parent_hash,
-        } = self.step
-        {
+        if let Some((target_priority_info, expected_parent_hash)) = self.step.generation_completed() {
```

Comment:
Could you check this? It has a problem if the `proposal_generated` event is called later, this code meets assertion error.

Reply:
I removed the assertion because the generation time is not guaranteed. I added traces messages for the cases where the block generated in different rounds.